frontend/src/components/Dialog.js

```diff
@@ -38,4 +39,47 @@ const SendMessageButton = ({
   )
 }
 
-export default SendMessageButton
+export const RoundButton = ({
+  setIsOpen,
+  isOpen,
+  dialogRef,
+  imagePath,
+  optionalMargin = '0px',
+  optionalImageHeight = 'auto',
+  setQRContent,
+  contentKey,
+}) => {
+  const handleOnClick = async () => {
+    setIsOpen(!isOpen)
+    if (!isOpen) {
+      dialogRef.current?.show()
+    } else {
+      dialogRef.current?.close()
+    }
+
+    const qrContent = await getQRContentToOnboard(contentKey)
+
+    setQRContent(qrContent)
+    await approveMember(contentKey) //approves member
+  }
```

Comment:
could move this function out of the component and pass it in - to keep it generic